backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -192,9 +193,7 @@ void presentProof(@NonNull PresentationExchangeRecord presentationExchangeRecord
                                 String msg = "No matching credentials found for proof request: "
                                         + presentationExchangeRecord.getPresentationExchangeId();
                                 log.warn(msg);
-                                pProofRepo.findByPresentationExchangeId(
-                                        presentationExchangeRecord.getPresentationExchangeId())
-                                        .ifPresent(pp -> pProofRepo.updateProblemReport(pp.getId(), msg));
+                                throw new PresentationConstructionException(msg);
```

Comment:
This disables the feature that highlights the exchange that has issues in the frontend. We can discuss if we need both. An immediate response for the user who triggered the exchange, and a record of the failed exchange for future reference or other users.

Reply:
This is the method that executes the users click to respond to a proof request. Without throwing  an exception the endpoint responds with a 200, when it actually failed, so throwing an exception seems standard. I'll do both though just in case it's used somewhere else.